pkg/operators/cli/clioperator.go

```diff
@@ -241,15 +245,34 @@ func (o *cliOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error
 			// TODO: compatiblity for now: add all; remove me later on and use the commented version above
 			opts = []json.Option{json.WithShowAll(true)}
 
+			if o.mode == ModeJSONPretty {
+				opts = append(opts, json.WithPretty(true, "  "))
+			}
+
 			jsonFormatter, err := json.New(ds, opts...)
 			if err != nil {
 				return fmt.Errorf("initializing JSON formatter: %w", err)
 			}
 
-			ds.Subscribe(func(ds datasource.DataSource, data datasource.Data) error {
+			df := func(ds datasource.DataSource, data datasource.Data) error {
 				fmt.Println(string(jsonFormatter.Marshal(data)))
 				return nil
-			}, Priority)
+			}
+
+			if o.mode == ModeYAML {
+				// For the time being, this uses a slow approach to marshal to YAML, by first
+				// converting to JSON and then to YAML. This should get a dedicated formatter sooner or later.
+				df = func(ds datasource.DataSource, data datasource.Data) error {
+					yml, err := yaml.JSONToYAML(jsonFormatter.Marshal(data))
+					if err != nil {
+						return fmt.Errorf("serializing yaml: %w", err)
+					}
+					fmt.Println("---")
```

Comment:
It's not strictly related to this PR, but just want to highlight that the CLI operator still needs to start using the `frontends/console` pkg.